plugin-k8s/src/lib.rs

```diff
@@ -84,69 +104,153 @@ impl AlumetPlugin for K8sPlugin {
+                                    // Case of an other extension than .slice is found --> I will not find cpu.stat file
+                                    return;
+                                }
+                            },
+                        };
+                        if let Some(pod_uid) = path.file_name() {
+                            let pod_uid = match pod_uid.to_str() {
+                                Some(uid_str) => uid_str,
+                                None => {
+                                    log::error!("Unable to transform to str");
+                                    return;
+                                }
+                            };
                             // We open a File Descriptor to the newly created file
                             let mut path_cpu = path.clone();
+                            let full_name_to_seek = pod_uid.strip_suffix(".slice").unwrap_or(&pod_uid);
+                            let parts: Vec<&str> = full_name_to_seek.split("pod").collect();
+                            let name_to_seek_raw = *(parts.last().unwrap_or(&full_name_to_seek));
+                            let uid_raw = parts.last().unwrap_or(&"No UID found");
+                            let uid = format!("pod{}", uid_raw);
+                            let name_to_seek = name_to_seek_raw.replace("_", "-");
+                            // let (name, ns) = cgroup_v2::get_pod_name(name_to_seek.to_owned());
+                            let hostname_ostring = gethostname();
+                            let hostname = hostname_ostring
+                                .to_str()
+                                .context("Invalid UTF-8 in Hostname")
+                                .unwrap_or("")
+                                .to_string();
+                            let rt = match tokio::runtime::Builder::new_current_thread().enable_all().build() {
+                                Ok(rt_ok) => rt_ok,
+                                Err(err) => {
+                                    log::error!("Runtime failed to build and returned an error: {}", err);
+                                    return;
+                                }
+                            };
+                            let (name, namespace, node) = match rt.block_on(async {
+                                cgroup_v2::get_pod_name(
+                                    name_to_seek.to_owned(),
+                                    hostname,
+                                    self.kubernetes_api_url.clone(),
+                                )
+                                .await
+                            }) {
+                                Ok(tuple_found) => tuple_found,
+                                Err(err) => {
+                                    log::error!("Block on failed returned an error: {}", err);
+                                    return;
+                                }
+                            };
```

Comment:
`.context("...")?` or `.with_context(|| format!("..."))?` would be more readable than a match